app/main.py

```diff
@@ -1,3 +1,17 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
-    pass
+
+    repository = {}
+
+    @functools.wraps(func)
+    def wrapper(*args):
+        if args not in repository:
+            result = func(*args)
+            repository.update({args: result})
+            print("Calculating new result")
```

Comment:
Function should always return a result. Now it just prints a message in this branch